File: TaxiApp/FrontEnd-App/src/app/shared/model/cars/car.model.ts
import { JsonDateConverter } from "@app/shared/json-mapper";
import { JsonObject, JsonProperty } from "json2typescript";

@JsonObject
export class CarModel {

    @JsonProperty("id", Number)
	public id: number = null;

    @JsonProperty("model", String)
	public model: string = null;

    @JsonProperty("registrationNumber", String)
	public registrationNumber: string = null;

    @JsonProperty("lastTechControl", JsonDateConverter)
	public lastTechControl: Date = null;

    @JsonProperty("driverId", Number)
	public driverId: number = null;

}

File: TaxiApp/src/ro/taxiApp/docs/presentation/client/shared/model/cars/CarModel.java
package ro.taxiApp.docs.presentation.client.shared.model.cars;

import java.util.Date;

public class CarModel {

	private Long id;
	private String model;
	private String registrationNumber;
	private Date lastTechControl;
	
	private Long driverId;
	private Long carCategoryId;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getModel() {
		return model;
	}

	public void setModel(String model) {
		this.model = model;
	}

	public String getRegistrationNumber() {
		return registrationNumber;
	}

	public void setRegistrationNumber(String registrationNumber) {
		this.registrationNumber = registrationNumber;
	}

	public Date getLastTechControl() {
		return lastTechControl;
	}

	public void setLastTechControl(Date lastTechControl) {
		this.lastTechControl = lastTechControl;
	}

	public Long getDriverId() {
		return driverId;
	}

	public void setDriverId(Long driverId) {
		this.driverId = driverId;
	}

	public Long getCarCategoryId() {
		return carCategoryId;
	}

	public void setCarCategoryId(Long carCategoryId) {
		this.carCategoryId = carCategoryId;
	}
	
}
